lldb/source/Plugins/ObjectFile/Minidump/MinidumpFileBuilder.cpp

```diff
@@ -858,10 +923,224 @@ Status MinidumpFileBuilder::Dump(lldb::FileUP &core_file) const {
   return error;
 }
 
-size_t MinidumpFileBuilder::GetDirectoriesNum() const {
-  return m_directories.size();
+Status MinidumpFileBuilder::AddMemoryList_32(
+    const Process::CoreFileMemoryRanges &ranges) {
+  std::vector<MemoryDescriptor> descriptors;
+  Status error;
+  Log *log = GetLog(LLDBLog::Object);
+  size_t region_index = 0;
```

Comment:
Make on DataBufferHeap outside of this loop and re-use it:
```
lldb::offset_t max_size = 0;
for (const auto &core_range : ranges) {
  if (core_range.renge.size() > max_size)
    max_size = core_range.renge.size();
}
auto data_up = std::make_unique<DataBufferHeap>(max_size, 0);
```

auto data_up = std::make_unique<DataBufferHeap>(size, 0);

Reply:
Done. Albeit the `m_data` buffer that the class uses does get used and then reallocated when we call `Clear()` in flush to disk. May want to look into keeping a write index and a range...